src/main.cc

```diff
@@ -932,8 +933,7 @@ mainReconfigureFinish(void *)
         }
     } catch (...) {
         // for now any errors are a fatal condition...
-        debugs(1, DBG_CRITICAL, "FATAL: Unhandled exception parsing config file. " <<
-               " Run squid -k parse and check for errors.");
+        debugs(1, DBG_CRITICAL, "FATAL: Configuration error: " << CurrentException);
```

Comment:
Ah. I see buggy code. These catch were supposed to be the handlers for all config parse exceptions because there is different needs for startup and reconfigure cases. It would probably be best to drop these changes for now and we fix the exception handling in a followup PR where we can debate the best way to resolve it.

Reply:
> I see buggy code from ba98516.

FYI: That trivial commit does not seem to be related to this discussion.

Reply:
> It would probably be best to drop these changes for now

Agreed. These fixes are out of this PR scope. If they are not trivial, let's not piggyback them.

Reply:
I have opened #776 for the exception handling.

> FYI: That trivial commit does not seem to be related to this discussion.

Sorry, copied the hash from github 'blame' report rather than the page link. Correct hash is 4eac3407573794dec7f40816dc1f1f3139c2e63e .

Reply:
> Please restore the mention of -k parse

Restored in commits 0584f47 and e13a611.